enterprise/backend/src/metabase_enterprise/advanced_permissions/query_processor/middleware/permissions.clj

```diff
@@ -92,24 +95,25 @@
               (apply min (filter some? [original-limit max-rows-in-limited-downloads])))
     query))
 
-(defn limit-download-result-rows
+(defenterprise limit-download-result-rows
   "Post-processing middleware to limit the number of rows included in downloads if the user has `limited` download
   perms. Mainly useful for native queries, which are not modified by the [[apply-download-limit]] pre-processing
   middleware."
+  :feature :advanced-permissions
   [query rff]
   (if (and (is-download? query)
-           (premium-features/has-feature? :advanced-permissions)
            (= (current-user-download-perms-level query) :limited))
```

Comment:
why remove the `premium-features/has-feature?` here but not in `apply-download-limit` or `check-download-permissions`?

Reply:
Yeah, we should probably do it in those places too, we don't need it anymore because `defenterprise` does it automatically